### DISCUSSION TITLE
Control system power button behavior

### DISCUSSION THREAD

**Discussion by mattie47** [labwc/labwc#2370] [Q&A] [answered]
Hi,

I'm using a RPI with a physical latching power button. When it's latched the Pi should be on, when it's unlatched, the Pi should trigger the poweroff process. 

When I press the button to turn the Pi off, systemd-logind sends a `KEY_POWER` event to listeners such as x11/wayfire/wayland. 

```
Nov 21 22:58:12 sdm-test systemd-logind[590]: Power key pressed short.
```

In attempting to switch from wayfire/X11 to labwc, I can't seem to figure out a way of emulating the same behavior of this single event automatically triggering a shutdown process. 

Right now I'm finding I have to latch/unlatch the power button at least twice for "Logout/Restart/Shutdown" menu to appear and then trigger a shutdown process.

In X11, unlatching the power button will trigger the shutdown process.

In Wayfire, I can add modify `~/.config/wayfire.ini` from `command_power=pwrkey` to `command_power=systemctl poweroff`

Does labwc support a method of listening to power events and then triggering an action such as automatically shutting down?

Input Events:
https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h#L191 

Thanks,

Matt

**Comment by ahesford** [comment]
Using a tool like `wev` will tell you what name is associated with this key. You can then use that in your labwc key bindings to do what you want.

**Reply by mattie47** [reply]
Unfortunately `wev` and `xev` don't seem to pick up the shutdown key press. 

`showkey` does though:

```
pi@sdm-test:~ $ sudo showkey
kb mode was ?UNKNOWN?
[ if you are trying this under X, it might not work
since the X server is also reading /dev/console ]

press any key (program terminates 10s after last keypress)...
keycode 116 press
```

**Comment by stefonarch** [comment, marked as answer]
In LXQt we use:
```
    <!-- Powerbutton-->
    <keybind key="XF86PowerOff">
      <action name="Execute">
        <command>lxqt-leave</command>
      </action>
    </keybind>
```

**Reply by mattie47** [reply]
Thank you! That worked!

I thought it may have been some form of XF86Power.... but I couldn't bruteforce the syntax. I had gone hunting for a list of `XF86<...>` names but couldn't find a complete list.

For anyone interested, this is what I ended up with in my `/home/pi/.config/labwc/rc.xml` file (you can likely ignore the touch device config)

```
<?xml version="1.0"?>
<openbox_config xmlns="http://openbox.org/3.4/rc">
        <touch deviceName="10-0038 generic ft5x06 (79)" mapToOutput="DSI-1" mouseEmulation="yes"/>


<labwc_config>

 <keyboard>
    <numlock>on</numlock>
    <layoutScope>global</layoutScope>
    <repeatRate>25</repeatRate>
    <repeatDelay>600</repeatDelay>
    <keybind key="A-Tab">
      <action name="NextWindow" />
    </keybind>

    <!-- Powerbutton-->
    <keybind key="XF86PowerOff">
      <action name="Execute">
        <command>systemctl poweroff</command>
      </action>
    </keybind>
 </keyboard>

</labwc_config>

</openbox_config>
```


Thanks again for providing that help and answer!

**Reply by mattie47** [reply]
Also just to update, I now see labwc default configs for RPI are stored in `/etc/xdg/labwc/rc.xml`


We see by default it has the following configuration:

```
    <keybind key="XF86PowerOff" onRelease="yes">
      <action name="Execute">
        <command>pwrkey</command>
      </action>
    </keybind>
```
